sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -184,7 +184,7 @@ RetT *waitUntilBuilt(KernelProgramCache &Cache,
 ///         cache. Accepts nothing. Return pointer to built entity.
 template <typename RetT, typename ExceptionT, typename KeyT, typename AcquireFT,
           typename GetCacheFT, typename BuildFT>
-RetT *getOrBuild(KernelProgramCache &KPCache, const KeyT &CacheKey,
+RetT *getOrBuild(KernelProgramCache &KPCache, KeyT &&CacheKey,
```

Comment:
Why do you need universal reference for `CacheKey` here?

Reply:
to avoid copying of the CacheKey when working with the cache map, as the key now includes the spec constant set.